### ISSUE TITLE
Add certificate logging option

### ISSUE THREAD

**Issue by michaelmaillot** [pnp/action-cli-login#12]
Hello,

I've never worked on developing Actions so I don't know if it's possible but... I would like to improve this one by adding options to allow logging through a certificate path (not sure about this one) or encoded entries.

In a business context, I think it would be relevant to automatically login with credentials others than user / password.

What do you think? If it's fine, I can give it a try 🤗

**Comment by garrytrinder** [comment]
Great suggestion @michaelmaillot we should absolutely include different options for authentication to Microsoft 365 to suit different scenarios.

I think a quick way to implement certificate login is we could add a new action input that accepts a string pass the value to the `--certificateBase64Encoded` option in the `login` command. The certificate can then be stored easily in a GitHub secret and accessed by the workflow when executed.

**Comment by michaelmaillot** [comment]
That was also what I though! Comparing to Azure DevOps, there's no way apparently to store files such as certificates so passing by encoded parameters stored as secrets might be the solution.

If you don't mind, I'll try to work on this.

**Comment by garrytrinder** [comment]
> If you don't mind, I'll try to work on this.

Thank you, it's all yours! 👍

**Comment by michaelmaillot** [comment]
Hey @garrytrinder, I wanted to push the feature branch, to test it out, but looks like I'm not allowed to do so (403 error).

Would you mind allowing me required permissions?

**Comment by waldekmastykarz** [comment]
Is there a way to test it from your fork? To keep things clean and manageable I'd prefer that we avoid work branches in this repo if possible

**Comment by michaelmaillot** [comment]
Sure, but what about pushing the branch in order to submit a PR?

**Comment by waldekmastykarz** [comment]
You can submit a PR from your fork, right?

**Comment by michaelmaillot** [comment]
Oh. Well... I forgot to fork the repo. Which makes the forbidden error understandable.

Really sorry, I'm such a mindless man... 🤦‍♂️😄

**Comment by waldekmastykarz** [comment]
All good, always good to check twice. We're in this together

**Comment by garrytrinder** [comment]
I've just merged #16 so we can now look to progress this PR.

@michaelmaillot I've made some changes to the codebase to make it much more testable therefore there will be some updates that need to be made, also note that the `master` branch has been renamed to `main`, to update your local clone execute the below git command.

```sh
git branch -m master main
git fetch origin
git branch -u origin/main main
git remote set-head origin -a
```
